### PR TITLE
Update Razor Components and Blazor templates

### PR DESCRIPTION
**Pull request by SteveSandersonMS**
... to use the latest package names and startup APIs.

### CODE REVIEW

File: src/Components/blazor/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted-CSharp/BlazorHosted-CSharp.Client/BlazorHosted-CSharp.Client.csproj [hunk 1]
```diff
@@ -4,14 +4,16 @@
...
     </RestoreAdditionalProjectSources>
     <LangVersion>7.3</LangVersion>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Components.Browser" Version="$(TemplateBlazorPackageVersion)" />
-    <PackageReference Include="Microsoft.AspNetCore.Components.Build" Version="$(TemplateBlazorPackageVersion)" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.AspNetCore.Blazor" Version="$(TemplateBlazorPackageVersion)" />
+    <PackageReference Include="Microsoft.AspNetCore.Blazor.Build" Version="$(TemplateBlazorPackageVersion)" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.NET.Sdk.Razor" Version="$(MicrosoftNETSdkRazorPackageVersion)" PrivateAssets="All" />
```
Comment (rynowak) [inline src/Components/blazor/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted-CSharp/BlazorHosted-CSharp.Client/BlazorHosted-CSharp.Client.csproj]: Do you expect this to be present in all of these? What this is needed for is when you want to pin the Razor SDK to a specific version. If you expect the user to have an SDK that supports components, this shouldn't be needed.

File: src/Components/blazor/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted-CSharp/BlazorHosted-CSharp.Server/BlazorHosted-CSharp.Server.csproj [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
     </RestoreAdditionalProjectSources>
     <LangVersion>7.3</LangVersion>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.App" />
-    <PackageReference Include="Microsoft.AspNetCore.Components.Server" Version="$(TemplateBlazorPackageVersion)" />
-    <PackageReference Include="Microsoft.AspNetCore.Razor.Design" Version="$(TemplateRazorDesignPackageVersion)" />
+    <PackageReference Include="Microsoft.AspNetCore.Blazor.Server" Version="$(TemplateBlazorPackageVersion)" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.Server" Version="$(TemplateComponentsPackageVersion)" />
```
Comment (rynowak) [inline src/Components/blazor/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted-CSharp/BlazorHosted-CSharp.Server/BlazorHosted-CSharp.Server.csproj]: For instance this doesn't reference the Razor SDK

File: src/Components/blazor/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted-CSharp/BlazorHosted-CSharp.Client/BlazorHosted-CSharp.Client.csproj (hunk 1 above)
Comment (SteveSandersonMS) [reply src/Components/blazor/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted-CSharp/BlazorHosted-CSharp.Client/BlazorHosted-CSharp.Client.csproj]: There are weird compiler warnings about duplicated types if we don't pin the Razor SDK to a specific version here (i.e., if I remove L16).

Are you saying we should trust the SDK will eventually give us the right version?

Similarly, do you know if we know yet when there will be an end-to-end working preview of what's shipping in Preview 2? I'd like to be able to verify that the .NET CLI tooling, VS2019, the ASP.NET runtime, and our templates all work together without any warnings/errors.

**Review by rynowak — APPROVED** [review]

File: src/Components/blazor/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted-CSharp/BlazorHosted-CSharp.Client/BlazorHosted-CSharp.Client.csproj (hunk 1 above)
Comment (SteveSandersonMS) [reply src/Components/blazor/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted-CSharp/BlazorHosted-CSharp.Client/BlazorHosted-CSharp.Client.csproj]: We discussed, and the outcome is that we do expect L16 to go away before we ship, but we do need it to stay there temporarily.

**Pull request merged**